src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/TopicProducer.cs

```diff
@@ -137,6 +130,25 @@ public async Task Send(ProducerContext<TKey, TValue> context)
                 }
             }
 
+            async Task<Message<byte[], byte[]>> CreateMessage(TopicPartition topic, KafkaSendContext<TKey, TValue> sendContext)
+            {
+                var headers = _context.HeadersSerializer.Serialize(sendContext);
```

Comment:
Perhaps move this after the two Tasks but before the Task.WhenAll().

Reply:
it is impossible as it requires for Key/Value serialization